src/connectors/hits-per-page/connectHitsPerPage.js

```diff
@@ -229,19 +226,12 @@ You may want to add another entry to the \`items\` option with this value.`
       },
 
       getWidgetSearchParameters(searchParameters, { uiState }) {
-        const hitsPerPage = uiState.hitsPerPage;
-        if (hitsPerPage)
-          return searchParameters.setQueryParameter(
-            'hitsPerPage',
-            uiState.hitsPerPage
-          );
-        if (defaultValue) {
-          return searchParameters.setQueryParameter(
-            'hitsPerPage',
-            defaultValue.value
-          );
-        }
-        return searchParameters.setQueryParameter('hitsPerPage', undefined);
+        const hitsPerPage =
+          uiState.hitsPerPage || (defaultItem && defaultItem.value);
+
+        return searchParameters.setQueryParameters({
+          hitsPerPage,
```

Comment:
IMO the widget should be deterministic. Here it's not the case, when we don't provide a default value we don't know how many hits per page we'll have. It depends on the context where the widget is added. I think that we should fallback on the first item when the default value is not provided or enforce the default value (this is what we do inside React InstantSearch).

Reply:
That's a change in the way the widget acts.

I'll be in favor of having another option that determines the default value as well. Should we go for `defaultValue`?

See [React InstantSearch reference to `defaultRefinement`](https://www.algolia.com/doc/api-reference/widgets/hits-per-page/react/#widget-param-defaultrefinement)

Reply:
We already have a mechanism for the default value. We don't have to introduce another option. We have to either enforce it or find a fallback. If you really want to introduce another option we can but I don't see the value.

Reply:
The value would be a less confusing API. Right now it's easy to miss or to add several `default` values. Then we get into a tricky situation of making a choice for the user to select the right one. If we add an option to select the default one, it's more straightforward for users.

I think this should be part of another PR. Let's not diverge too much here.

Reply:
See #4069.